### ISSUE TITLE
csharp_space_around_binary_operators requires space before binary operator immediately proceeding an opening parentheses

### ISSUE THREAD

**Issue by Arthri** [dotnet/roslyn#70095]
**Version Used**: 4.6.0-3.23259.8 (c3cc1d0c). .NET 7.0.304

**Steps to Reproduce**:
1. Write the following to a .editorconfig file beside the project
    ```editorconfig
    [*.cs]
    root = true
    dotnet_diagnostic.IDE0055.severity = error
    csharp_space_around_binary_operators = before_and_after
    ```
1. Write the following to a C# source file
    ```cs
    _ = (1, 1) is (< 1, < 1);
    ```
1. Run `dotnet format --verify-no-changes`

**Diagnostic Id**: IDE0055

**Expected Behavior**: No errors.

**Actual Behavior**: `Program.cs(1,16): error WHITESPACE: Fix whitespace formatting. Insert '\s'.`. Insert space where the dot is `_ = (1, 1) is (`·`< 1, < 1);`

**Comment by CyrusNajmabadi** [comment]
Tracking with https://github.com/dotnet/roslyn/issues/31691
